src/states/node.rs

```diff
@@ -110,6 +110,8 @@ pub struct Node {
     routing_msg_backlog: Vec<RoutingMessage>,
     /// Cache of `OwnSectionMerge` messages we have received, by sender section prefix.
     merge_cache: LruCache<Prefix<XorName>, SectionMap>,
+    /// Cache of union of our merged section, deduced from multiple `OwnSectionMerge`.
+    our_merged_section_cache: LruCache<Prefix<XorName>, (bool, BTreeSet<XorName>)>,
```

Comment:
I think the decision from the discussion was to always only keep at most one entry in this cache, i.e. make it an `Option` instead of `LruCache`?